content/en/blog/_posts/2024-07-31-consistent-read-from-cache.md

```diff
@@ -0,0 +1,87 @@
+---
+layout: blog
+title: 'Kubernetes Consistent Reads from Cache: A Leap in Performance and Scalability'
+date: 2024-07-31
+slug: consistent-read-from-cache-beta
+author: >
+  Marek Siarkowicz (Google)
+---
+
+Kubernetes is renowned for its robust orchestration of containerized applications,
+but as clusters grow, the demands on the control plane can become a bottleneck.
+A key challenge has been ensuring strongly consistent reads from the etcd datastore,
+requiring resource-intensive quorum reads.
+
+Today, the Kubernetes community is excited to announce a major improvement:
+**Consistent Reads from Cache**, graduating to Beta in Kubernetes 1.31.
+
+### Why consistent reads matter
+
+Consistent reads are essential for ensuring that Kubernetes components have an accurate view of the latest cluster state.
+Guaranteeing consistent reads is crucial for maintaining the accuracy and reliability of Kubernetes operations,
+enabling components to make informed decisions based on up-to-date information.
+In large-scale clusters, fetching and processing this data can be a performance bottleneck,
+especially for requests that involve filtering results.
+While Kubernetes can filter data by namespace directly within etcd,
+any other filtering by labels or field selectors requires the entire dataset to be fetched from etcd and then filtered in-memory by the Kubernetes API server.
+This is particularly impactful for components like the kubelet,
+which only needs to list pods scheduled on its node but forces the API Server and etcd to process all pods in the cluster.
+
+### The breakthrough: Caching with confidence
+
+Kubernetes has long used a watch cache to optimize read operations.
+The watch cache stores a snapshot of the cluster state and receives updates through etcd watches.
+However, until now, it couldn't serve consistent reads directly, as there was no guarantee the cache was sufficiently up-to-date.
+
+The **Consistent Reads from Cache** feature addresses this by leveraging etcd's "progress notifications."
```

Comment:
nit: IMO we can add a link to `progress notifications` in etcd docs, for someone who wants to read more about this feature.

Reply:
There is no nice holistic progress notification documentation on etcd page. The only thing I found is https://etcd.io/docs/v3.5/dev-guide/interacting_v3/ which is not very comprehensive.